f1180d5... Add "defbutton=true" to dialogs.  Fixes ENTER doing default action: n#439222.
63b7689... Michael's new table-based find and replace.
84991df... Apply Michael's boxtest patch.

diff --git sc/uiconfig/layout/insert-sheet.xml sc/uiconfig/layout/insert-sheet.xml
index 0ce5843..517a24d 100644
--- sc/uiconfig/layout/insert-sheet.xml
+++ sc/uiconfig/layout/insert-sheet.xml
@@ -49,7 +49,7 @@
 	    </vbox>
 	</hbox>
 	<dialogbuttonhbox border="5" spacing="5">
-            <okbutton default="true" id="BTN_OK"/>
+            <okbutton defbutton="true" id="BTN_OK"/>
             <cancelbutton id="BTN_CANCEL"/>
             <helpbutton id="BTN_HELP"/>
 	</dialogbuttonhbox>
diff --git sc/uiconfig/layout/move-copy-sheet.xml sc/uiconfig/layout/move-copy-sheet.xml
index 79720df..6e68b5a 100644
--- sc/uiconfig/layout/move-copy-sheet.xml
+++ sc/uiconfig/layout/move-copy-sheet.xml
@@ -15,7 +15,7 @@
 	    <flow/>
             <cancelbutton id="BTN_CANCEL" tabstop="true"/>
             <helpbutton id="BTN_HELP" tabstop="true"/>
-            <okbutton default="true" id="BTN_OK" tabstop="true"/>
+            <okbutton defbutton="true" id="BTN_OK" tabstop="true"/>
 	</dialogbuttonhbox>
     </vbox>
 </modaldialog>
diff --git sc/uiconfig/layout/string-input.xml sc/uiconfig/layout/string-input.xml
index 0a59899..a29a564 100644
--- sc/uiconfig/layout/string-input.xml
+++ sc/uiconfig/layout/string-input.xml
@@ -8,7 +8,7 @@
         <fixedtext id="FT_LABEL" label="FT-LABEL"/>
         <edit has_border="true" id="ED_INPUT" tabstop="true"/>
 	<dialogbuttonhbox border="5" spacing="5">
-            <okbutton default="true" id="BTN_OK" tabstop="true"/>
+            <okbutton defbutton="true" id="BTN_OK" tabstop="true"/>
             <cancelbutton id="BTN_CANCEL" tabstop="true"/>
             <helpbutton id="BTN_HELP" tabstop="true"/>
 	</dialogbuttonhbox>
diff --git svx/uiconfig/layout/find-and-replace.xml svx/uiconfig/layout/find-and-replace.xml
index ad5af2b..9b922a7 100644
--- svx/uiconfig/layout/find-and-replace.xml
+++ svx/uiconfig/layout/find-and-replace.xml
@@ -4,69 +4,69 @@
      transex3/layout/tralay.  !-->
 
 <sfxmodelessdialog _title="Find &amp; Replace" help-id="SID_SEARCH_DLG" closeable="true" sizeable="true" moveable="true" output-size="true" id="RID_SVXDLG_SEARCH" sv-look="true" xmlns="http://openoffice.org/2007/layout" xmlns:cnt="http://openoffice.org/2007/layout/container">
-    <vbox spacing="5" border="5">
+    <table cnt:spacing="5" cnt:border="5" columns="2" homogeneous="false">
 	<string id="RID_SVXSTR_SEARCH_STYLES" _text="Including Styles"/>
 	<string id="RID_SVXSTR_SEARCH" _text="(Search)"/>
 	<string id="RID_SVXSTR_REPLACE" _text="(Replace)"/>
 	<string id="STR_WORDCALC" _text="~Entire cells"/>
 	<string id="STR_MORE_BTN" _text="More ~Options"/>
 	<string id="STR_LESS_BTN" _text="Less ~Options"/>
-	<hbox homogeneous="false" spacing="5" border="0">
-	    <vbox spacing="5" border="0">
-		<fixedtext id="FT_SEARCH" _label="~Search for"/>
-		<combobox dropdown="true" id="ED_SEARCH" tabstop="true" cnt:expand="false"/>
-		<listbox has_border="true" dropdown="true" show="false" id="LB_SEARCH" sort="true" tabstop="true"/>
-		<fixedtext show="false" id="FT_SEARCH_ATTR" word-break="true"/>
-		<fixedtext show="false" id="FT_SEARCH_FORMATS" word-break="true"/>
-		<flow/>
-		<flow/>
-		<flow/>
-	    </vbox>
-	    <vbox cnt:expand="false" spacing="5" border="0">
-		<min-size min-width="100" cnt:expand="false"/>
-		<pushbutton default="true" id="BTN_SEARCH" tabstop="true" _label="~Find" cnt:expand="false"/>
-		<pushbutton id="BTN_SEARCH_ALL" tabstop="true" _label="Find ~All" cnt:expand="false"/>
-	    </vbox>
-	</hbox>
-        <hbox spacing="5" border="0">
-	    <flow/>
-	    <vbox cnt:expand="false" spacing="5" border="0">
-		<min-size min-width="100" cnt:expand="false"/>
-		<fixedline id="FL_SEARCH_COMMAND"/>
-	    </vbox>
-	</hbox>
-	<hbox homogeneous="false" spacing="5" border="0">
-	    <vbox spacing="5" border="0">
-		<fixedtext id="FT_REPLACE" _label="Re~place with"/>
-		<combobox dropdown="true" id="ED_REPLACE" tabstop="true" cnt:expand="false"/>
-		<listbox has_border="true" dropdown="true" show="false" id="LB_REPLACE" sort="true" tabstop="true"/>
-		<fixedtext show="false" id="FT_REPLACE_ATTR" word-break="true"/>
-		<fixedtext show="false" id="FT_REPLACE_FORMATS" word-break="true"/>
-		<fixedline show="false" id="FL_SEARCH_COMPONENT"/>
-		<flow/>
-		<flow/>
-		<flow/>
-	    </vbox>
-	    <vbox cnt:expand="false" spacing="5" border="0">
-		<min-size min-width="100" cnt:expand="false"/>
-		<pushbutton id="BTN_REPLACE" tabstop="true" _label="~Replace" cnt:expand="false"/>
-		<pushbutton id="BTN_REPLACE_ALL" tabstop="true" _label="Replace A~ll" cnt:expand="false"/>
-	    </vbox>
-	</hbox>
-        <checkbox id="CB_MATCH_CASE" tabstop="true" _label="Ma~tch case"/>
-        <pushbutton show="false" id="BTN_COMPONENT_1" tabstop="true" cnt:expand="false"/>
-        <checkbox id="CB_WHOLE_WORDS" tabstop="true" _label="Whole wor~ds only"/>
-        <pushbutton show="false" id="BTN_COMPONENT_2" tabstop="true" cnt:expand="false"/>
-        <fixedline id="FL_BUTTONS"/>
-	<hbox>
+
+	<!-- row -->
+	<fixedtext id="FT_SEARCH" _label="~Search for"/>
+	<imagebutton defbutton="true" id="BTN_SEARCH" tabstop="true" _label="~Find" cnt:expand="false"
+		     image-align="0" graphic="res/commandimagelist/lc_searchdialog.png"/>
+	<!-- row -->
+	<vbox spacing="0" border="2" cnt:x-expand="true">
+	     <combobox dropdown="true" id="ED_SEARCH" tabstop="true" cnt:expand="true"/>
+	     <listbox has_border="true" dropdown="true" show="false" id="LB_SEARCH" sort="true" tabstop="true" cnt:expand="true"/>
+	</vbox>
+	<pushbutton id="BTN_SEARCH_ALL" tabstop="true" _label="Find ~All" cnt:expand="false"/>
+	<!-- row -->
+	<vbox spacing="0" border="0" cnt:col-span="2" cnt:y-expand="false">
+	    <fixedtext show="false" id="FT_SEARCH_ATTR" word-break="true"/>
+	    <fixedtext show="false" id="FT_SEARCH_FORMATS" word-break="true"/>
+	</vbox>
+	<!-- row -->
+	<fixedline cnt:y-expand="false" cnt:col-span="2"/>
+	<!-- row -->
+	<fixedtext id="FT_REPLACE" _label="Re~place with"/>
+	<pushbutton id="BTN_REPLACE" tabstop="true" _label="~Replace" cnt:expand="false"/>
+	<!-- row -->
+	<vbox spacing="0" border="5" cnt:x-expand="true">
+	    <combobox dropdown="true" id="ED_REPLACE" tabstop="true" cnt:expand="false"/>
+	    <listbox has_border="true" dropdown="true" show="false" id="LB_REPLACE" sort="true" tabstop="true"/>
+	</vbox>
+	<pushbutton id="BTN_REPLACE_ALL" tabstop="true" _label="Replace A~ll" cnt:expand="false"/>
+	<!-- row -->
+	<vbox spacing="5" border="0" cnt:col-span="2">
+	    <fixedtext show="false" id="FT_REPLACE_ATTR" word-break="true"/>
+	    <fixedtext show="false" id="FT_REPLACE_FORMATS" word-break="true"/>
+	    <fixedline show="false" id="FL_SEARCH_COMPONENT"/>
+	</vbox>
+	<!-- row -->
+	<vbox spacing="5" border="0" cnt:col-span="2">
+	    <flow>
+                <checkbox id="CB_MATCH_CASE" tabstop="true" _label="Ma~tch case"/>
+                <pushbutton show="false" id="BTN_COMPONENT_1" tabstop="true" cnt:expand="false"/>
+                <checkbox id="CB_WHOLE_WORDS" tabstop="true" _label="Whole wor~ds only"/>
+                <pushbutton show="false" id="BTN_COMPONENT_2" tabstop="true" cnt:expand="false"/>
+	    </flow>
+	</vbox>
+	<!-- row -->
+        <fixedline cnt:y-expand="false" cnt:col-span="2"/>
+	<!-- row -->
+	<hbox cnt:col-span="2">
             <morebutton delta="174" map-unit="MAP_APPFONT" id="BTN_MORE" state="false" tabstop="true"/>
             <dialogbuttonhbox>
 		<helpbutton id="BTN_HELP" tabstop="true" cnt:expand="false"/>
 		<cancelbutton id="BTN_CLOSE" tabstop="true" _label="~Close" cnt:expand="false"/>
             </dialogbuttonhbox>
 	</hbox>
-        <fixedline show="false" id="FL_OPTIONS"/>
-	<hbox homogeneous="false" spacing="5" border="0">
+	<!-- row -->
+        <fixedline show="false" id="FL_OPTIONS" cnt:col-span="2"/>
+	<!-- row -->
+	<hbox homogeneous="false" spacing="5" border="0" cnt:col-span="2">
 	    <vbox spacing="5" border="0">
 		<checkbox show="false" id="CB_SELECTIONS" tabstop="true" _label="Current selection ~only"/>
 		<checkbox show="false" id="CB_BACKWARDS" tabstop="true" _label="Bac~kwards"/>
@@ -107,5 +107,5 @@
 		<pushbutton show="false" id="BTN_NOFORMAT" tabstop="true" _label="~No Format" cnt:expand="false"/>
 	    </vbox>
 	</hbox>
-    </vbox>
+    </table>
 </sfxmodelessdialog>
diff --git svx/uiconfig/layout/zoom.xml svx/uiconfig/layout/zoom.xml
index 5938643..9f9d782 100644
--- svx/uiconfig/layout/zoom.xml
+++ svx/uiconfig/layout/zoom.xml
@@ -51,9 +51,9 @@
         <fixedline cnt:padding="1" id="FL_BOTTOM"/>
 	<dialogbuttonhbox border="5" spacing="5">
 	    <flow/>
-	    <okbutton     id="BTN_ZOOM_OK"/>
+	    <okbutton defbutton="true" id="BTN_ZOOM_OK"/>
 	    <cancelbutton id="BTN_ZOOM_CANCEL"/>
-	    <helpbutton   id="BTN_ZOOM_HELP"/>
+	    <helpbutton id="BTN_ZOOM_HELP"/>
 	</dialogbuttonhbox>
     </vbox>
 </modaldialog>
diff --git sw/uiconfig/layout/wordcount.xml sw/uiconfig/layout/wordcount.xml
index e922294..fbab778 100644
--- sw/uiconfig/layout/wordcount.xml
+++ sw/uiconfig/layout/wordcount.xml
@@ -43,7 +43,7 @@
         <fixedline cnt:padding="1" id="FL_BOTTOM"/>
 	<dialogbuttonhbox spacing="5">
             <flow/>
-            <okbutton id="PB_OK"/>
+            <okbutton defbutton="true" id="PB_OK"/>
             <helpbutton id="PB_HELP"/>
         </dialogbuttonhbox>
     </vbox>
diff --git toolkit/source/layout/core/box-base.hxx toolkit/source/layout/core/box-base.hxx
index 4fa7e53..e430e22 100644
--- toolkit/source/layout/core/box-base.hxx
+++ toolkit/source/layout/core/box-base.hxx
@@ -57,7 +57,7 @@ public:
     struct ChildProps: public PropHelper
     {
         //ChildProps( ChildProps* );
-        PROPHELPER_SET_INFO
+	PROPHELPER_SET_INFO
     };
 
 protected:
diff --git toolkit/source/layout/core/box.hxx toolkit/source/layout/core/box.hxx
index 98190ca..5a0e870 100644
--- toolkit/source/layout/core/box.hxx
+++ toolkit/source/layout/core/box.hxx
@@ -61,6 +61,7 @@ public:
     struct ChildProps : public Box_Base::ChildProps
     {
         ChildProps( ChildData *pData );
+	PROPHELPER_SET_INFO
     };
 
 protected:
diff --git toolkit/source/layout/core/table.hxx toolkit/source/layout/core/table.hxx
index dac2aaa..f1c882b 100644
--- toolkit/source/layout/core/table.hxx
+++ toolkit/source/layout/core/table.hxx
@@ -58,6 +58,7 @@ public:
     struct ChildProps : public Box_Base::ChildProps
     {
         ChildProps( ChildData *pData );
+	PROPHELPER_SET_INFO
     };
 
 protected:
diff --git toolkit/src2xml/source/src2xml.py toolkit/src2xml/source/src2xml.py
index 128cfa1..b03c5e6 100644
--- toolkit/src2xml/source/src2xml.py
+++ toolkit/src2xml/source/src2xml.py
@@ -182,7 +182,7 @@ def post_process (s):
     s = s.replace ('<modaldialog', '<modaldialog sizeable="true"')
     s = s.replace (' rid=', ' id=')
     s = s.replace (' border="true"', ' has_border="true"')
-    s = s.replace (' def-button="true"', ' default="true"')
+    s = s.replace (' def-button="true"', ' defbutton="true"')
     s = s.replace (' drop-down="', ' dropdown="')
     s = s.replace (' tab-stop="', ' tabstop="')
     return s
diff --git toolkit/uiconfig/layout/tab-dialog.xml toolkit/uiconfig/layout/tab-dialog.xml
index 1d631e9..2639752 100644
--- toolkit/uiconfig/layout/tab-dialog.xml
+++ toolkit/uiconfig/layout/tab-dialog.xml
@@ -9,7 +9,7 @@
 	<tabcontrol id="ID_TABCONTROL"/>
 	<dialogbuttonhbox border="5" spacing="5" cnt:expand="false">
 	    <flow/>
-	    <okbutton id="BTN_OK" show="false"/>
+	    <okbutton defbutton="true" id="BTN_OK" show="false"/>
 	    <pushbutton id="BTN_USER" _label="&lt;User&gt;" show="false"/>
 	    <cancelbutton id="BTN_CANCEL" show="false"/>
 	    <helpbutton id="BTN_HELP" show="false"/>
diff --git toolkit/workben/layout/TEST toolkit/workben/layout/TEST
old mode 100644
new mode 100755
index d6a20cb..8a04a66
--- toolkit/workben/layout/TEST
+++ toolkit/workben/layout/TEST
@@ -1,3 +1,11 @@
-export OOO_INSTALL_PREFIX=$(cd ../../../../.. && pwd)/ooo-svn
+#! /bin/sh
+for d in $(cd ../../../../.. && pwd)/ooo-svn /usr/lib/ooo3/; do
+    if test -d $d; then
+        export OOO_INSTALL_PREFIX=$d
+	break
+    fi
+done
+export SAL_ALLOW_LINKOO_SYMLINKS=1
 test -d $OOO_INSTALL_PREFIX/openoffice.org3 && export OOO_PREFIX_BRAND=$OOO_INSTALL_PREFIX/openoffice.org3 || export OOO_PREFIX_BRAND=$OOO_INSTALL_PREFIX
-LD_LIBRARY_PATH=$OOO_PREFIX_BRAND/basis-link/program:$OOO_PREFIX_BRAND/basis-link/ure-link/lib ../../unxlngx6.pro/bin/test "$@"
+export LD_LIBRARY_PATH="$OOO_PREFIX_BRAND/basis-link/program:$OOO_PREFIX_BRAND/basis-link/ure-link/lib"
+../../unx*.pro/bin/test "$@"
diff --git toolkit/workben/layout/boxtest.xml toolkit/workben/layout/boxtest.xml
index 8015b3f..263ae40 100644
--- toolkit/workben/layout/boxtest.xml
+++ toolkit/workben/layout/boxtest.xml
@@ -4,6 +4,8 @@
         xmlns:cnt="http://openoffice.org/2007/layout/container"
         title="Box Test" optimumsize="true" has_border="true" sizeable="true" moveable="true">
 	<vbox>
+	  <table>
+	    <vbox>
 		<hbox cnt:expand="false" cnt:fill="false">
 			<pushbutton label="XML Left"
 			            cnt:expand="false" cnt:fill="false"/>
@@ -28,5 +30,7 @@
 	                <pushbutton label="XML Right"
 	                            cnt:expand="true" cnt:fill="true"/>
 	        </hbox>
+		</vbox>
+	</table>
 	</vbox>
 </dialog>
diff --git toolkit/workben/layout/editor.cxx toolkit/workben/layout/editor.cxx
index b5e5d76..3fbc889 100644
--- toolkit/workben/layout/editor.cxx
+++ toolkit/workben/layout/editor.cxx
@@ -344,9 +344,14 @@ public:
             {
                 beans::Property prop = it.next();
                 rtl::OUString name( prop.Name );
-                rtl::OUString value( pChild->getProperty( name, CONTAINER_PROPERTY ) );
-                std::pair< rtl::OUString, rtl::OUString > pair( name, value );
-                pChild->maOriChildProps.push_back( pair );
+		try {
+			rtl::OUString value( pChild->getProperty( name, CONTAINER_PROPERTY ) );
+			std::pair< rtl::OUString, rtl::OUString > pair( name, value );
+			pChild->maOriChildProps.push_back( pair );
+		} catch ( beans::UnknownPropertyException &rEx ) {
+			fprintf (stderr, "ERROR: widget reports that it has a property it cannot return: '%s' this normally means that someone screwed up their PROPERTY_SET_INFO macro usage.\n",
+				 rtl::OUStringToOString (rEx.Message, RTL_TEXTENCODING_UTF8).getStr());
+		}
             }
         }
 
diff --git toolkit/workben/layout/insert-sheet.xml toolkit/workben/layout/insert-sheet.xml
index 0ce5843..517a24d 100644
--- toolkit/workben/layout/insert-sheet.xml
+++ toolkit/workben/layout/insert-sheet.xml
@@ -49,7 +49,7 @@
 	    </vbox>
 	</hbox>
 	<dialogbuttonhbox border="5" spacing="5">
-            <okbutton default="true" id="BTN_OK"/>
+            <okbutton defbutton="true" id="BTN_OK"/>
             <cancelbutton id="BTN_CANCEL"/>
             <helpbutton id="BTN_HELP"/>
 	</dialogbuttonhbox>
diff --git toolkit/workben/layout/move-copy-sheet.xml toolkit/workben/layout/move-copy-sheet.xml
index 79720df..6e68b5a 100644
--- toolkit/workben/layout/move-copy-sheet.xml
+++ toolkit/workben/layout/move-copy-sheet.xml
@@ -15,7 +15,7 @@
 	    <flow/>
             <cancelbutton id="BTN_CANCEL" tabstop="true"/>
             <helpbutton id="BTN_HELP" tabstop="true"/>
-            <okbutton default="true" id="BTN_OK" tabstop="true"/>
+            <okbutton defbutton="true" id="BTN_OK" tabstop="true"/>
 	</dialogbuttonhbox>
     </vbox>
 </modaldialog>
diff --git toolkit/workben/layout/plugin.xml toolkit/workben/layout/plugin.xml
index f630e72..908907d 100644
--- toolkit/workben/layout/plugin.xml
+++ toolkit/workben/layout/plugin.xml
@@ -20,7 +20,7 @@
 	</hbox>
 	<dialogbuttonhbox border="5" spacing="5">
 	    <flow/>
-	    <okbutton id="BTN_OK"/>
+	    <okbutton defbutton="true" id="BTN_OK"/>
 	    <cancelbutton id="BTN_CANCEL" />
 	    <helpbutton id="BTN_HELP"/>
 	</dialogbuttonhbox>
diff --git toolkit/workben/layout/recover.xml toolkit/workben/layout/recover.xml
index 7c96191..1d74b59 100644
--- toolkit/workben/layout/recover.xml
+++ toolkit/workben/layout/recover.xml
@@ -58,7 +58,7 @@ LINE TEST"/>
 	<checkbox id="CH_LOGFILE" _label="Open additional text document to display the restore logfile"/>
 	<dialogbuttonhbox border="5" spacing="5">
 	    <flow/>
-	    <okbutton id="BTN_OK" _label="Restore"/>
+	    <okbutton defbutton="true" id="BTN_OK" _label="Restore"/>
 	    <cancelbutton id="BTN_CANCEL" />
 	    <helpbutton id="BTN_HELP"/>
 	</dialogbuttonhbox>
diff --git toolkit/workben/layout/sortdlg.xml toolkit/workben/layout/sortdlg.xml
index a17920e..ce45a60 100644
--- toolkit/workben/layout/sortdlg.xml
+++ toolkit/workben/layout/sortdlg.xml
@@ -56,7 +56,7 @@
             </pagelist>
         </tabcontrol>
         <hbox>
-            <okbutton id="1" tabstop="true"/>
+            <okbutton defbutton="true" id="1" tabstop="true"/>
             <cancelbutton id="1" tabstop="true"/>
             <helpbutton id="1" tabstop="true"/>
             <pushbutton id="1" tabstop="true"/>
diff --git toolkit/workben/layout/string-input.xml toolkit/workben/layout/string-input.xml
index 0a59899..a29a564 100644
--- toolkit/workben/layout/string-input.xml
+++ toolkit/workben/layout/string-input.xml
@@ -8,7 +8,7 @@
         <fixedtext id="FT_LABEL" label="FT-LABEL"/>
         <edit has_border="true" id="ED_INPUT" tabstop="true"/>
 	<dialogbuttonhbox border="5" spacing="5">
-            <okbutton default="true" id="BTN_OK" tabstop="true"/>
+            <okbutton defbutton="true" id="BTN_OK" tabstop="true"/>
             <cancelbutton id="BTN_CANCEL" tabstop="true"/>
             <helpbutton id="BTN_HELP" tabstop="true"/>
 	</dialogbuttonhbox>
diff --git toolkit/workben/layout/tab-dialog.xml toolkit/workben/layout/tab-dialog.xml
index 1d631e9..2639752 100644
--- toolkit/workben/layout/tab-dialog.xml
+++ toolkit/workben/layout/tab-dialog.xml
@@ -9,7 +9,7 @@
 	<tabcontrol id="ID_TABCONTROL"/>
 	<dialogbuttonhbox border="5" spacing="5" cnt:expand="false">
 	    <flow/>
-	    <okbutton id="BTN_OK" show="false"/>
+	    <okbutton defbutton="true" id="BTN_OK" show="false"/>
 	    <pushbutton id="BTN_USER" _label="&lt;User&gt;" show="false"/>
 	    <cancelbutton id="BTN_CANCEL" show="false"/>
 	    <helpbutton id="BTN_HELP" show="false"/>
diff --git toolkit/workben/layout/wordcount-plain.xml toolkit/workben/layout/wordcount-plain.xml
index cf31a76..a7c34fe 100644
--- toolkit/workben/layout/wordcount-plain.xml
+++ toolkit/workben/layout/wordcount-plain.xml
@@ -22,7 +22,7 @@
         </hbox>
         <fixedline height="8" id="FL_BOTTOM" width="158" x="6" y="77"/>
         <hbox>
-            <okbutton def-button="true" height="14" id="PB_OK" width="50" x="61" y="88"/>
+            <okbutton defbutton="true" height="14" id="PB_OK" width="50" x="61" y="88"/>
             <helpbutton height="14" id="PB_HELP" width="50" x="114" y="88"/>
         </hbox>
     </vbox>
diff --git toolkit/workben/layout/wordcount-tight.xml toolkit/workben/layout/wordcount-tight.xml
index 9ff84b3..b027510 100644
--- toolkit/workben/layout/wordcount-tight.xml
+++ toolkit/workben/layout/wordcount-tight.xml
@@ -28,7 +28,7 @@
         <fixedline cnt:padding="1" id="FL_BOTTOM"/>
         <hbox border="5">
 	    <flow homogeneous="true"/>
-	    <okbutton id="PB_OK" cnt:expand="false" default="true"/>
+	    <okbutton id="PB_OK" cnt:expand="false" defbutton="true"/>
 	    <helpbutton id="PB_HELP" cnt:expand="false"/>
         </hbox>
     </vbox>
diff --git toolkit/workben/layout/wordcount.xml toolkit/workben/layout/wordcount.xml
index e922294..fbab778 100644
--- toolkit/workben/layout/wordcount.xml
+++ toolkit/workben/layout/wordcount.xml
@@ -43,7 +43,7 @@
         <fixedline cnt:padding="1" id="FL_BOTTOM"/>
 	<dialogbuttonhbox spacing="5">
             <flow/>
-            <okbutton id="PB_OK"/>
+            <okbutton defbutton="true" id="PB_OK"/>
             <helpbutton id="PB_HELP"/>
         </dialogbuttonhbox>
     </vbox>
diff --git toolkit/workben/layout/zoom-indent.xml toolkit/workben/layout/zoom-indent.xml
index 5a6a226..648da86 100644
--- toolkit/workben/layout/zoom-indent.xml
+++ toolkit/workben/layout/zoom-indent.xml
@@ -33,7 +33,7 @@
 	    </hbox>
 	</vbox>
 	<vbox border="6" spacing="4">
-	    <okbutton     id="OkButton"     cnt:expand="false" default="true"/>
+	    <okbutton     id="OkButton"     cnt:expand="false" defbutton="true"/>
 	    <cancelbutton id="CancelButton" cnt:expand="false"/>
 	    <helpbutton   id="HelpButton"   cnt:expand="false"/>
 	</vbox>
diff --git toolkit/workben/layout/zoom-plain.xml toolkit/workben/layout/zoom-plain.xml
index 30b72a3..c8383b8 100644
--- toolkit/workben/layout/zoom-plain.xml
+++ toolkit/workben/layout/zoom-plain.xml
@@ -3,7 +3,7 @@
 <modaldialog sizeable="true" height="142" help-id="SID_ATTR_ZOOM" moveable="true" output-size="true" id="RID_SVXDLG_ZOOM" sv-look="true" text="Zoom" width="160" xmlns="http://openoffice.org/2007/layout" xmlns:cnt="http://openoffice.org/2007/layout/container">
     <vbox>
         <fixedline height="8" id="FL_ZOOM" text="Zoom factor" width="92" x="6" y="3"/>
-        <okbutton def-button="true" height="14" id="BTN_ZOOM_OK" width="50" x="104" y="6"/>
+        <okbutton defbutton="true" height="14" id="BTN_ZOOM_OK" width="50" x="104" y="6"/>
         <radiobutton height="10" id="BTN_WHOLE_PAGE" width="80" x="12" y="14"/>
         <cancelbutton height="14" id="BTN_ZOOM_CANCEL" width="50" x="104" y="23"/>
         <radiobutton height="10" id="BTN_PAGE_WIDTH" width="80" x="12" y="27"/>
diff --git toolkit/workben/layout/zoom.xml toolkit/workben/layout/zoom.xml
index 5938643..9f9d782 100644
--- toolkit/workben/layout/zoom.xml
+++ toolkit/workben/layout/zoom.xml
@@ -51,9 +51,9 @@
         <fixedline cnt:padding="1" id="FL_BOTTOM"/>
 	<dialogbuttonhbox border="5" spacing="5">
 	    <flow/>
-	    <okbutton     id="BTN_ZOOM_OK"/>
+	    <okbutton defbutton="true" id="BTN_ZOOM_OK"/>
 	    <cancelbutton id="BTN_ZOOM_CANCEL"/>
-	    <helpbutton   id="BTN_ZOOM_HELP"/>
+	    <helpbutton id="BTN_ZOOM_HELP"/>
 	</dialogbuttonhbox>
     </vbox>
 </modaldialog>
